### PR TITLE
adding high precision stopwatch possibilities

### PR DESCRIPTION
**Pull request by KjellKod**
This adds a non-breaking API change. 

the default StopWatch is now an alias for `ChronoMeter<>`, or `ChronoMeter<std::chrono::steady_clock>`

It is also possible to use PrecisionStopWatch, which is an alias for `ChronoMeter<std:chrono::high_precision_clock>`

### CODE REVIEW

**Comment by john-gress** [comment]
Looks like there might be a problem with this change in Research:
```
libtool: compile:  /usr/local/probe/bin/g++ -DPACKAGE_NAME=\"nm\" -DPACKAGE_TARNAME=\"nm\" -DPACKAGE_VERSION=\"0.1\" "-DPACKAGE_STRING=\"nm 0.1\"" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"nm\" -DVERSION=\"0.1\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DBUILD_VERSION=\"0.1.1\" -DHAVE_IOSTREAM=1 -I. -I. -I./libfork -I./libheartbeat -I./libnetwork -I./liblrdpi -I./libsyslog -I./libcapture -I./libcommand -I./libprocess -I./liblua -I./libelasticsearch -I./libtools -I./libsystem -I./libmachineid -I/usr/local/probe/include -include /usr/include/fenv.h -Werror -Wall -DINSTALL_PREFIX=\"/home/jgress/wk/Research\" -DSPLIT_QUEUES -Wl,--no-as-needed -L/usr/local/probe/lib -L/usr/local/probe/lib64 -Wl,-rpath -Wl,/usr/local/probe/lib64 -Wl,-rpath -Wl,/usr/local/probe/lib -Ofast -m64 -std=c++14 -MT libcapture/libcapture_libcapture_la-PacketCaptureReceiver.lo -MD -MP -MF libcapture/.deps/libcapture_libcapture_la-PacketCaptureReceiver.Tpo -c libcapture/PacketCaptureReceiver.cpp  -fPIC -DPIC -o libcapture/.libs/libcapture_libcapture_la-PacketCaptureReceiver.o
In file included from ProbeLogger.cpp:16:0:
./libsyslog/SyslogReporter.h:28:7: error: using typedef-name ‘using StopWatch = class ChronoMeter<>’ after ‘class’
 class StopWatch;
       ^
In file included from ./liblua/LuaChangeSubscriber.h:10:0,
                 from ./liblua/LuaExecuter.h:8,
                 from ./liblrdpi/RuleEngine.h:13,
                 from ProbeLogger.cpp:15:
/usr/local/probe/include/StopWatch.h:78:32: note: ‘using StopWatch = class ChronoMeter<>’ has a previous declaration here
 using StopWatch = ChronoMeter<>;
                                ^
```

**Comment by KjellKod** [comment]
@john-gress thanks John. I'll put up a PR for Research to remove the offending line and also make sure there are not other offending lines of the same issue.

**Comment by KjellKod** [comment]
@john-gress please see Research PR that takes care of this: https://github.schq.secious.com/Logrhythm/Research/pull/1708

**Comment by john-gress** [comment]
👍

File: src/StopWatch.h [hunk 1]
```diff
@@ -1,34 +1,81 @@
...
+    * Resets the start point
+    * @return the updated start point
+    */
+   typename ChronoType::time_point Restart() {
+      mStart = clock::now();
+      return mStart;
+   }
+
+ protected:
+   typename clock::time_point mStart;
```
Comment (shwetabhandare) [inline src/StopWatch.h]: Why isn't this variable declared as private?

**Comment by shwetabhandare** [comment]
I have one question in the PR.
I pulled this down  and the Research PR and works well. 👍

File: src/StopWatch.h (hunk 1 above)
Comment (KjellKod) [inline src/StopWatch.h]: don't know. it's lost in time somewhere. I didn't change that.   I'm suspecting that currently or earlier it was used for a mock.  I'm not planning to address that here.

**Pull request merged**